### PR TITLE
#12330 Fix logic of disable months in month picker

### LINKED ISSUE #12330: Calendar: wrong logic of disabling the month
### Describe the bug

If we enter min date greater then the first day of month, we won't be able to choose that month in the "month view" of calendar.
Example: minDate = 02.12.2022, December is not able to select
image
image

### Environment

- macOS: 13.0

### Reproducer

https://stackblitz.com/edit/primeng-calendar-demo-s6vvzw?file=src/app/app.component.ts

### Angular version

14.0.0

### PrimeNG version

14.2.1

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

16.17.0

### Browser(s)

Google Chrome 107.0

### Steps to reproduce the behavior

_No response_

### Expected behavior

Even if one day is selectable, so month is selectable also

### PR DESCRIPTION
**Pull request by RTcyan**
Fixed #12330

### Defect Fixes
Fixed following issue: 
[Calendar: wrong logic of disabling the month #12330](https://github.com/primefaces/primeng/issues/12330)

### CODE REVIEW

File: src/app/components/calendar/calendar.spec.ts [hunk 1]
```diff
@@ -588,6 +588,35 @@ describe('Calendar', () => {
         expect(calendar.inputFieldValue).toEqual('01');
     }));
 
+    fit('should use min date and max date with month picker', fakeAsync(() => {
```
Comment (volvachev) [inline src/app/components/calendar/calendar.spec.ts:591]: remove 'f' - should be `it`

Comment (RTcyan) [reply src/app/components/calendar/calendar.spec.ts]: Yes, thank you! My mistake

**Review by cetincakiroglu — APPROVED** [review]

**Comment by cetincakiroglu** [comment]
Hi @RTcyan,

Good catch, could you please update your pr to pass all checks of pr checker? I see that your format failed please refactor by using "npm run format" command

**Comment by cetincakiroglu** [comment]
Hi @RTcyan,

I see that pr checker still throwing error due to code formatting. Next time please use "npm run format" command before pushing the PR so we can merge it safely.

Thanks for your effort and support!

**Pull request merged**